server/hotrod/src/test/java/org/infinispan/server/hotrod/HotRodReplicationTest.java

```diff
@@ -112,18 +112,19 @@ public void testReplicatedPutWithTopologyChanges(Method m) {
 
       resp = clients().get(1).put(k(m), 0, 0, v(m, "v2-"), (byte) 2, 0);
       assertStatus(resp, Success);
-      assertTopologyReceived(resp.topologyResponse, servers(), currentServerTopologyId());
+      int serverClientTopologyId = currentServerTopologyId();
+      assertTopologyReceived(resp.topologyResponse, servers(), serverClientTopologyId);
 
       resp = clients().get(0)
-                      .put(k(m), 0, 0, v(m, "v3-"), (byte) 2, ClusterCacheStatus.INITIAL_TOPOLOGY_ID + 2 * nodeCount());
+            .put(k(m), 0, 0, v(m, "v3-"), (byte) 2, serverClientTopologyId);
       assertStatus(resp, Success);
       assertEquals(resp.topologyResponse, null);
       assertSuccess(clients().get(1).get(k(m), 0), v(m, "v3-"));
 
       HotRodServer newServer = startClusteredServer(servers().get(1).getPort() + 25);
       try {
          resp = clients().get(0)
-                         .put(k(m), 0, 0, v(m, "v4-"), (byte) 2, ClusterCacheStatus.INITIAL_TOPOLOGY_ID + 2 * nodeCount());
+               .put(k(m), 0, 0, v(m, "v4-"), (byte) 2, serverClientTopologyId);
          assertStatus(resp, Success);
          assertEquals(resp.topologyResponse.topologyId, currentServerTopologyId());
```

Comment:
I guess this is what is missing for the test to work
```suggestion
         serverClientTopologyId = currentServerTopologyId();
         assertEquals(resp.topologyResponse.topologyId, serverClientTopologyId);
```

Reply:
Yes, this is the problem, good catch. Updated below as well.